crates/core/tedge/src/cli/mqtt/cli.rs

```diff
@@ -44,41 +43,16 @@ pub enum TEdgeMqttCli {
 
 impl BuildCommand for TEdgeMqttCli {
     fn build_command(self, context: BuildContext) -> Result<Box<dyn Command>, crate::ConfigError> {
-        let port = context
-            .config_repository
-            .load()?
-            .query(MqttClientPortSetting)?;
-        let host = context
-            .config_repository
-            .load()?
-            .query(MqttClientHostSetting)?;
-        let ca_file = context
-            .config_repository
-            .load()?
-            .query(MqttClientCafileSetting)
-            .ok();
-        let ca_path = context
-            .config_repository
-            .load()?
-            .query(MqttClientCapathSetting)
-            .ok();
-        let client_cert = context
-            .config_repository
-            .load()?
-            .query(MqttClientAuthCertSetting);
-        let client_private_key = context
-            .config_repository
-            .load()?
-            .query(MqttClientAuthKeySetting);
+        let config = context.config_repository.load_new()?;
+        let client_cert = config.mqtt.client.auth.ca_file.or_none().cloned();
+        let client_private_key = config.mqtt.client.auth.key_file.or_none().cloned();
 
-        let client_auth_config = if client_cert.is_err() && client_private_key.is_err() {
+        let client_auth_config = if client_cert.is_none() && client_private_key.is_none() {
             None
         } else {
-            let client_cert = client_cert?;
-            let client_private_key = client_private_key?;
             Some(ClientAuthConfig {
-                cert_file: client_cert,
-                key_file: client_private_key,
+                cert_file: client_cert.unwrap_or_default(),
+                key_file: client_private_key.unwrap_or_default(),
```

Comment:
It's kinda related to what @jarhodes314 explained why `auth` is private.

The code has a problem. If `client_cert` is `Some` and `client_private_key` is `None`, it enters this `else` conditional branch. Actually, it should have been already an error for that case. But this code allows to have such config.
```rust
client_auth_config = Some(ClientAuthConfig {
    cert_file: <some value>,
    client_private_key: <default of "Utf8PathBuf">
})
```

In short, you need to address the pair of `client_cert` and `client_private_key` as James suggested.

Reply:
fixed